site/app/controllers/grading/ImagesController.php

```diff
@@ -48,4 +54,139 @@ public function viewImagesPage() {
+            return $this->core->getOutput()->renderResultMessage("Empty POST request. This may mean that the sum size of your files are greater than {$max_size}.", false, false);
+        }
+
+        if (!isset($_POST['csrf_token']) || !$this->core->checkCsrfToken($_POST['csrf_token'])) {
+            return $this->core->getOutput()->renderResultMessage("Invalid CSRF token.", false, false);
+        }
+
+        $uploaded_files = array();
+        if (isset($_FILES["files1"])) {
+            $uploaded_files[1] = $_FILES["files1"];
+        }
+        $errors = array();
+        $count_item = 0;
+        if (isset($uploaded_files[1])) {
+            $count_item = count($uploaded_files[1]["name"]);
+            for ($j = 0; $j < $count_item[1]; $j++) {
+                if (!isset($uploaded_files[1]["tmp_name"][$j]) || $uploaded_files[1]["tmp_name"][$j] === "") {
+                    $error_message = $uploaded_files[1]["name"][$j]." failed to upload. ";
+                    if (isset($uploaded_files[1]["error"][$j])) {
+                        $error_message .= "Error message: ". ErrorMessages::uploadErrors($uploaded_files[1]["error"][$j]). ".";
+                    }
+                    $errors[] = $error_message;
+                }
+            }
+        }
+
+        if (count($errors) > 0) {
+            $error_text = implode("\n", $errors);
+            return $this->core->getOutput()->renderResultMessage("Upload Failed: ".$error_text, false);
+        }
+
+        if (empty($uploaded_files)) {
+            return $this->core->getOutput()->renderResultMessage("No files to be submitted.", false);
+        }
+
+        $file_size = 0;
+        if (isset($uploaded_files[1])) {
+            $uploaded_files[1]["is_zip"] = array();
+            for ($j = 0; $j < $count_item; $j++) {
+                if (FileUtils::getMimeType($uploaded_files[1]["tmp_name"][$j]) == "application/zip") {
+                    if(FileUtils::checkFileInZipName($uploaded_files[1]["tmp_name"][$j]) === false) {
+                        return $this->core->getOutput()->renderResultMessage("Error: You may not use quotes, backslashes or angle brackets in your filename for files inside ".$uploaded_files[1]["name"][$j].".", false);
+                    }
+                    $uploaded_files[1]["is_zip"][$j] = true;
+                    $file_size += FileUtils::getZipSize($uploaded_files[1]["tmp_name"][$j]);
+                }
+                else {
+                    if(FileUtils::isValidFileName($uploaded_files[1]["name"][$j]) === false) {
+                        return $this->core->getOutput()->renderResultMessage("Error: You may not use quotes, backslashes or angle brackets in your file name ".$uploaded_files[1]["name"][$j].".", false);
+                    }
+                    $uploaded_files[1]["is_zip"][$j] = false;
+                    $file_size += $uploaded_files[1]["size"][$j];
+                }
+            }
+        }
+
+        $max_size = 10485760;
```

Comment:
Is it correct to add a file size limit here? This should probably be somewhere in the configuration.

Reply:
Yes they should probably not be here. After looking into it, I found it is a bit more complicated than several lines' changes. For this PR, I would like to keep it for router changes only. Please see #4008 for further discussion.